bedita-app/models/business/relation_check.php

```diff
@@ -0,0 +1,98 @@
+            $relationInverse = $relationData['inverse'];
+            $countAllowedTypesR = $relationStats->getObjectRelationsCount($relationInverse, $rightObjectTypes);
+            $countAllTypesR = $relationStats->getObjectRelationsCount($relationInverse);
+            $results[] = $this->checkTypesForRelation($relationInverse, $countAllowedTypesR, $countAllTypesR, $rightObjectTypes, $messageBuffer);
+            $results[] = $this->checkCounts('allowed types', $relationName, $relationInverse, $countAllowedTypesL, $countAllowedTypesR, $messageBuffer);
+            $results[] = $this->checkCounts('all types', $relationName, $relationInverse, $countAllTypesL, $countAllTypesR, $messageBuffer);
+        } else {
+            $messageBuffer[] = 'No inverse: no checks';
+        }
+        foreach ($results as $r) {
+            if ($r === false) {
+                $result = false;
+            }
+        }
+        $messageBuffer[] = ($result) ? 'ok' : 'corrupted data';
+        return $result;
+    }
+
+
+    private function checkTypesForRelation($relationName, $countRelAllowedTypes, $countRelAllTypes, $allowedTypes, &$messageBuffer) {
+        if ($countRelAllowedTypes != $countRelAllTypes) {
+            $messageBuffer[] = abs($countRelAllTypes - $countRelAllowedTypes) . ' records refer to a type not related to relation "' . $relationName . '" (MODEL-RELATED-TYPES-COUNT: ' . $countRelAllowedTypes . ' / ALL-TYPES-COUNT: ' . $countRelAllTypes . ')';
+            $relationStats = ClassRegistry::init('RelationStats');
+            $countRelationsGroupByType = $relationStats->getObjectRelationsCountGroupByType($relationName);
+            foreach ($countRelationsGroupByType as $objectTypeId => $data) {
+                if (!in_array($objectTypeId, $allowedTypes)) {
+                    $ot = Configure::read('objectTypes.'.$objectTypeId.'.name');
+                    $messageBuffer[] = '|=> ' . $data . ' records refer to the object type ' . $ot . ' (the object type is not related to relation ' . $relationName . ')';
+                }
+            }
+        }
+    }
+
+    private function checkCounts($info, $relationName, $relationInverse, $count1, $count2, &$messageBuffer) {
+        if ($count1 === $count2) {
+            $messageBuffer[] = "$info [left]->'$relationName'->[right]: $count1";
+            $messageBuffer[] = "$info [right]->'$relationInverse'->[left]: $count2";
+            return true;
+        } else if ($count1 < $count2) {
+            $messageBuffer[] = "$info [left]->'$relationName'->[right]: $count1 => should be $count2";
+            $messageBuffer[] = "$info [right]->'$relationInverse'->[left]: $count2";
+            return false;
+        } else {
+            $messageBuffer[] = "$info [left]->'$relationName'->[right]: $count1";
+            $messageBuffer[] = "$info [right]->'$relationInverse'->[left]: $count2 => should be $count1";
+            return false;
+        }
+        return true;
+    }
+}
+?>
```

Comment:
PHP files should ent with empty new line
http://www.php-fig.org/psr/psr-2/#files